### PR TITLE
sanitizing the URL

### PR DESCRIPTION
**Pull request by S-Panta**
Sanitizing the baseURL and addition of the integration test.
Related issue : https://github.com/owncloud/ocis-php-sdk/issues/24

### CODE REVIEW

File: tests/integration/Owncloud/OcisSdkPhp/OcisTest.php [hunk 1]
```diff
@@ -78,6 +78,18 @@ private function getAccessToken(string $username, string $password): string
         return $accessTokenResponse['access_token'];
     }
 
+    public function testListDrive(): void
```
Comment (amrita-shrestha) [inline tests/integration/Owncloud/OcisSdkPhp/OcisTest.php:81]:
```suggestion
    public function testListMyDrive(): void
```

File: tests/integration/Owncloud/OcisSdkPhp/OcisTest.php [hunk 2]
```diff
@@ -78,6 +78,18 @@ private function getAccessToken(string $username, string $password): string
...
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis('https://ocis.owncloud.test/', $token, ['verify' => false]);
+        $drives = $ocis->listMyDrives();
+        $driveName = [];
+        foreach ($drives as $drive) {
+            $driveName [] = $drive->getName();
+        }
+        $expectedDriveArray = ["Admin Admin","Shares"];
+        $this->assertEquals($expectedDriveArray, $driveName);
+    }
```
Comment (amrita-shrestha) [inline tests/integration/Owncloud/OcisSdkPhp/OcisTest.php:92]:
```suggestion
    }
    
```

File: tests/integration/Owncloud/OcisSdkPhp/OcisTest.php [hunk 3]
```diff
@@ -78,6 +78,14 @@ private function getAccessToken(string $username, string $password): string
         return $accessTokenResponse['access_token'];
     }
 
+    public function testListMyDriveResponseArrayForMalformedUrl(): void
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisSdkPhp/OcisTest.php:81]:
```suggestion
    public function testServiceUrlTrailingSlash(): void
```

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/integration/Owncloud/OcisSdkPhp/OcisTest.php [hunk 4]
```diff
@@ -78,6 +78,14 @@ private function getAccessToken(string $username, string $password): string
...
+    {
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis('https://ocis.owncloud.test///', $token, ['verify' => false]);
```
Comment (individual-it) [inline tests/integration/Owncloud/OcisSdkPhp/OcisTest.php:84]:
```suggestion
        $ocis = new Ocis($this->ocisUrl . '///', $token, ['verify' => false]);
```

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/output/clover.xml
Comment (phil-davis) [inline tests/output/clover.xml:1]: This should not be committed.

**Review by individual-it — APPROVED** [review]

**Pull request merged**